tasks/add_picklist_values.py

```diff
@@ -0,0 +1,339 @@
+import tempfile
+
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+from cumulusci.tasks.salesforce import Deploy
+
+package_xml_template = """<?xml version="1.0" encoding="UTF-8"?>
+<Package xmlns="http://soap.sforce.com/2006/04/metadata">
+    <types>
+        <members>{object}.{field}</members>
+        <name>CustomField</name>
+    </types>{record_types_block}
+    <version>45.0</version>
+</Package>"""
+
+package_xml_record_types_block_template = """
+    <types>{record_types}
+        <name>RecordType</name>
+    </types>"""
+
+package_xml_record_type_template = """
+        <members>{record_type_full_name}</members>"""
+
+object_template = """<?xml version="1.0" encoding="UTF-8"?>
+<CustomObject xmlns="http://soap.sforce.com/2006/04/metadata">
+    <fields>
+        <fullName>{field}</fullName>
+        <description>{description}</description>
+        <inlineHelpText>{helpText}</inlineHelpText>
+        <label>{label}</label>
+        <type>Picklist</type>
+        <valueSet>
+            <valueSetDefinition>
+                <sorted>{sorted}</sorted>{picklist_values}
+            </valueSetDefinition>
+        </valueSet>
+    </fields>{record_types}
+</CustomObject>"""
+
+picklist_value_template = """
+                <value>
+                    <fullName>{name}</fullName>
+                    <default>{default}</default>
+                    <label>{label}</label>
+                </value>"""
+
+record_type_picklist_template = """
+    <recordTypes>
+        <fullName>{name}</fullName>
+        <active>true</active>
+        <description>{description}</description>
+        <label>{label}</label>
+        <picklistValues>
+            <picklist>{field}</picklist>{picklist_values}
+        </picklistValues>
+    </recordTypes>"""
+
+record_type_picklist_value_template = """
+            <values>
+                <fullName>{name}</fullName>
+                <default>{default}</default>
+            </values>"""
+
+class AddPicklistValues(BaseSalesforceApiTask, Deploy):
+    task_options = {
+        "sobject": {
+            "description": "SObject of the picklist field being modified",
+            "required": True,
+        },
+        "field": {
+            "description": "The picklist field being modified",
+            "required": True,
+        },
+        "values": {
+            "description": "A comma-delimited list of the picklist values being added",
+            "required": True,
+        },
+        "recordtypes": {
+            "description": "The record types for which these new picklist values will be available",
+            "required": False,
+        },
+        "sorted": {
+            "description": "If true, sets the entire picklist to be sorted in alphabetical order",
+            "required": False,
+        },
+        "otherlast": {
+            "description": "If true, the Other value (if one exists) will remain at the end of the values",
+            "required": False,
+        }
+    }
+
```

Comment:
I'd recommend adding an `_init_options` method like this:

    def _init_options(self, kwargs):
        super()._init_options(kwargs)
        self.options["values"] = process_list_arg(self.options.get("values", []))
        self.options["recordtypes"] = process_list_arg(self.options.get("recordtypes", []))
        self.options["sorted"] = process_bool_arg(self.options.get("sorted", False))
        self.options["otherlast"] = process_bool_arg(self.options.get("otherlast", False))

Reasons to initialize options this way:

- It keeps the code in _run_task cleaner because you can assume the options are already initialized
- When the task is used in a flow, this runs while constructing the flow before any steps run, so errors in the options are caught sooner.
- Using the process_list_arg and process_bool_arg helpers (from cumulusci.core.util) makes sure the task can be passed lists and bools that have already been constructed (such as from options in cumulusci.yml) rather than just strings.